src/Glpi/Http/Listener/CheckGlpiConfigListener.php

```diff
@@ -0,0 +1,139 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Http\Listener;
+
+use Config;
+use DBConnection;
+use DBmysql;
+use Glpi\Config\LegacyConfigProviderListener;
+use Glpi\Http\Error\DisplayGlpiMisconfiguredPage;
+use Glpi\Http\ListenersPriority;
+use Session;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+final class CheckGlpiConfigListener implements EventSubscriberInterface
+{
+    private bool $skip_db_checks = false;
+
+    public function __construct(
+        #[Autowire('%kernel.project_dir%')] private readonly string $projectDir,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::REQUEST => [
+                ['onKernelRequest', ListenersPriority::LEGACY_LISTENERS_PRIORITIES[self::class]],
+                ['onAfterLegacyConfigurators', -10 + ListenersPriority::LEGACY_LISTENERS_PRIORITIES[LegacyConfigProviderListener::class]]
+            ],
+        ];
+    }
+
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        if (!$event->isMainRequest()) {
+            return;
+        }
+
+        $root_doc = $event->getRequest()->getBasePath();
+        $request_uri = $event->getRequest()->getRequestUri();
+
+        $this->skip_db_checks = false;
+        if ($event->getRequest()->server->has('REQUEST_URI')) {
+            if (preg_match('#^' . $root_doc . '/front/(css|locale).php#', $request_uri) === 1) {
+                $this->skip_db_checks  = true;
+            }
+
+            $no_db_checks_scripts = [
+                '#^' . $root_doc . '/$#',
+                '#^' . $root_doc . '/index.php#',
+                '#^' . $root_doc . '/install/install.php#',
+                '#^' . $root_doc . '/install/update.php#',
+            ];
+            foreach ($no_db_checks_scripts as $pattern) {
+                if (preg_match($pattern, $request_uri) === 1) {
+                    $this->skip_db_checks = true;
+                    break;
+                }
+            }
+        }
+
+        // Check if the DB is configured properly
+        if ($this->skip_db_checks) {
+            return;
+        }
+
+        if (!\is_file(GLPI_CONFIG_DIR . '/config_db.php')) {
+            Session::loadLanguage('', false);
+
+            $event->setResponse(new StreamedResponse(new DisplayGlpiMisconfiguredPage($this->projectDir), 500));
```

Comment:
The event propagation should be stopped here. We do not want other request listeners to be executed.

```suggestion
            $event->setResponse(new StreamedResponse($this->displayGlpiMisconfiguredPage(...), 500));
            $event->stopPropagation();
```

Reply:
Propagation is always stopped when a Response is added:

![image](https://github.com/user-attachments/assets/acea728f-ae37-42c1-972e-0bf7ae1c3059)


Also, the object is callable, and I might reuse it, so using a different callable isn't the best IMO. The goal of the callable object is to be able to be used elsewhere, and I'm not sure whether it's gonna be needed yet. If it's never gonna be used outside, I'll bring it back in the listener''s code